nomad/structs/services.go

```diff
@@ -808,6 +819,20 @@ func (s *Service) validateNomadService(mErr *multierror.Error) {
 	}
 }
 
+// validateIdentity performs validation on workload identity field populated by
+// the job mutating hook
+func (s *Service) validateIdentity() error {
+	if s.Identity != nil {
+		if s.Identity.Name != "" {
+			return fmt.Errorf("Service identity name must not be explicitly set in the jobspec")
```

Comment:
Hum....this is an interesting scenario, because there's only one point in which this is true: job registration.

After the job is registered, the _opposite_ is true: a service WI _must_ have a name (`consul-server/...`).

I wonder if it would be better to remove this check and only validate an empty name in the job registration hook?

Reply:
Couldn't we have both? I think it's still worth it to check that there's no pre-set WI service name upon registration, and then we could have another check in the job registration hook.